src/helics/core/ActionMessage.cpp

```diff
@@ -227,6 +227,32 @@ int ActionMessage::toByteArray (char *data, int buffer_size) const
     return actSize;
 }
 
+int ActionMessage::serializedByteCount () const
+{
+    // base size= 7 header fields(7*4 bytes)+flags(2 bytes)+counter(2 bytes)+time(8 bytes)+payload size(4
+    // bytes)+1byte for number of strings=45
+    static constexpr int base_size =
+      static_cast<int> (8 * sizeof (uint32_t) + 2 * sizeof (uint16_t) + sizeof (int64_t) + 1);
+
+    int size{base_size};
+    size += static_cast<int> (payload.size ());
+    // for time request add an additional 3*8 bytes
+    if (messageAction == CMD_TIME_REQUEST)
+    {
+        size += static_cast<int> (3 * sizeof (int64_t));
+    }
+    // add additional string data
+    if (!stringData.empty ())
+    {
+        for (auto &str : stringData)
+        {
+            // 4(to store the length)+length of the string
+            size += static_cast<int> (str.size ()) + 4;
```

Comment:
It might be slightly clearer if the +4 was changed to `sizeof (uint32_t)` so it matches the type used in the serialization function, and flipping the order of things in the comment or the line below so that they match..